documents/architectural-decisions/role-management.adoc

```diff
@@ -0,0 +1,45 @@
+== Role Management
+
+The XMPP Publish-Subscribe mechanism (XEP-0060) lacks an explicit description of how to implement role-based authentication for topics(nodes).
+It implies two possibilities on how to do so: collection nodes and the roster access model.
+
+The following aspects must be taken into account:
+
+* Preferably use existing XMPP protocols/mechanisms
+* The chosen strategy must be usable in most practical use-cases.
+
+=== Considered Options
+
+* Usage of Collection Nodes (XEP-0248)
+* Roster Access Model which supports groups (XEP-0144)
+* Custom Roles implemented on the Broker
+
+=== Decision Outcome
+
+Chosen option: Usage of Collection Nodes (XEP-0248), because the mechanism is the least intrusive that yet allows a powerful role-concept.
+We must, however, keep in mind that the XEP is deferred and might therefore not be fully supported by all XMPP servers.
+
+
+=== Pros and Cons of the Options
+
+==== Usage of Collection Nodes (XEP-0248)
+
+* Good, because existing XMPP mechanisms can be used.
+* Good, because with it few consumers can subscribe to many topics indirectly.
+* Good, because publishing permissions are managed on a topic level which encourages per-device-topics.
+* Good, because this mechanism is used in production.
+* Bad, because XEP-0248 is deferred
+
```

Comment:
Not sure I understand this 100% correctly, but isn't it necessary to structure the Collection Nodes in a specific way to support permission management? (-> This might be a Bad Thing™️)
e.g.

> Bad, because the Collection Nodes must be structured in a specific way to support access management

Reply:
@fabianhauser can you define "bad thing"? the one thing an administrator must understand is that a parent collection will serve all items of the containing nodes regardless of the permissions set in the node on which the original message was published. If someone does not get that, he/she has a problem. But i would not call the concept "bad" - or did I misunderstand your comment?

Reply:
@raphiz You are right, "bad" is probably not the right word 🙂 . What I meant is that with this decision, the administrator possibly has to structure Collection Nodes differently to represent "permissions", which might be perceived as limitation I guess?
(i.e., you represent two things with Collection Nodes - Collections and Grouped-Access-Permissions).
Or did i misunderstand the concept?

Reply:
Aaah, now I get what you mean.
It is true, this is kind of a limitation. However, those who run similar projects already in production will most likely use this as a mechanism for access management already - especially because there are many different publishers with very limited  and a few subscribers who basically access everything.
I guess i could elaborate this pro as well as the con here..